pylint/checkers/variables.py

```diff
@@ -1400,7 +1406,16 @@ def _check_consumer(
         if found_nodes is None:
             return (VariableVisitConsumerAction.CONTINUE, None)
         if not found_nodes:
-            self.add_message("used-before-assignment", args=node.name, node=node)
+            if node.name in current_consumer.consumed_uncertain:
+                confidence = CONTROL_FLOW
+            else:
+                confidence = HIGH
```

Comment:
@jacobtylerwalls You're the expert here, but are we sure we never use inference in any of the method used in control flow "inference"?

Reply:
Right, I wondered that, too. I'm sort of sure, but I'll take another look tomorrow.

Reply:
All I can see so far is inference being used for some of the cases for `undefined-variable`.

Reply:
I added a failing assertion in my local editable install of `astroid` to crash when inferring, and then pylinted some files from the test suite with --disable=all, --enable=used-before-assignment. I only saw some boilerplate inferences when loading the original ast trees or pylint itself.

Reply:
That brings up another point: do we want to use this for `undefined-variable` as well? And if so, do we want to change `CONTROL_FLOW` to also include inference? ie., `CONTROL_FLOW` represents control flow assumptions + inference.

Reply:
I added confidence `HIGH` for `undefined-variable`. I couldn't find any cases where `INFERRED` and `CONTROL_FLOW` are both used: the one edge case of `undefined-variable` that uses inference does not use control flow.

Reply:
Thanks!